From 16cd1159a4d3713ba87db1c8b9b669caa9231613 Mon Sep 17 00:00:00 2001
From: hjl <hjl@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 28 Jan 2015 19:39:42 +0000
Subject: [PATCH] Call release_input_file in claim_file_handler

	PR lto/64837
	* lto-plugin.c (release_input_file): New.
	(claim_file_handler): Call release_input_file.
	(onload): Set release_input_file.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@220212 138bc75d-0d04-0410-961f-82ee72b054a4
---
 lto-plugin/ChangeLog    | 7 +++++++
 lto-plugin/lto-plugin.c | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/lto-plugin/lto-plugin.c b/lto-plugin/lto-plugin.c
index 8d957402ba6..8e0a6579958 100644
--- a/lto-plugin/lto-plugin.c
+++ b/lto-plugin/lto-plugin.c
@@ -145,6 +145,7 @@ static ld_plugin_register_all_symbols_read register_all_symbols_read;
 static ld_plugin_get_symbols get_symbols, get_symbols_v2;
 static ld_plugin_register_cleanup register_cleanup;
 static ld_plugin_add_input_file add_input_file;
+static ld_plugin_release_input_file release_input_file;
 static ld_plugin_add_input_library add_input_library;
 static ld_plugin_message message;
 static ld_plugin_add_symbols add_symbols;
@@ -1006,6 +1007,8 @@ claim_file_handler (const struct ld_plugin_input_file *file, int *claimed)
   if (obj.objfile)
     simple_object_release_read (obj.objfile);
 
+  release_input_file (file);
+
   return LDPS_OK;
 }
 
@@ -1091,6 +1094,9 @@ onload (struct ld_plugin_tv *tv)
 	case LDPT_ADD_INPUT_FILE:
 	  add_input_file = p->tv_u.tv_add_input_file;
 	  break;
+	case LDPT_RELEASE_INPUT_FILE:
+	  release_input_file = p->tv_u.tv_release_input_file;
+	  break;
 	case LDPT_ADD_INPUT_LIBRARY:
 	  add_input_library = p->tv_u.tv_add_input_library;
 	  break;
-- 
2.17.1

